app/LDAP/LDAPActions.php

```diff
@@ -0,0 +1,80 @@
+<?php
+
+namespace App\LDAP;
+
+use App\Actions\User\Create;
+use App\Models\Configs;
+use App\Models\Logs;
+use App\Models\User;
+
+/**
+ * Class LDAPUpdateJobs.
+ *
+ * This class manages the update between the database and LDAP.
+ */
+class LDAPActions
+{
+	public function __construct()
+	{
+	}
+
+	/**
+	 * Create a user in the database if it does not exist already.
+	 */
+	public static function create_user_not_exist(string $username, LDAPUserData $userData): void
+	{
+		$user = User::query()->where('username', '=', $username)->first();
```

Comment:
use `->count()` instead.

`->first()` hydrates a `User` model which adds slow downs and increase the memory usage.